### PR TITLE
feat: make Moby version configurable

### LINKED ISSUE #107: Moby version should be configurable
See #106

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**Reason for Change**:

Moby version should be configurable in apimodel. 

**Issue Fixed**:

Fixes #107 

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: .pipelines/e2e-step-template.yaml [hunk 1]
```diff
@@ -25,3 +25,4 @@ steps:
     inputs:
       artifactName: '${{ parameters.job }}_logs'
       targetPath: '$(modulePath)/_logs'
+    condition: always()
```
[the remark this answers is not in the document]
Comment (CecileRobertMichon) [reply .pipelines/e2e-step-template.yaml]: #415

File: parts/k8s/kubernetesparams.t [hunk 2]
```diff
@@ -319,6 +319,18 @@
...
     },
+    "mobyVersion": {
+      "defaultValue": "3.0.1",
+      "metadata": {
+        "description": "The Azure Moby build version"
+      },
+      "allowedValues": [
+         "3.0.1",
+         "3.0.2",
+         "3.0.3"
```
[the remark this answers is not in the document]
Comment (jackfrancis) [reply parts/k8s/kubernetesparams.t]: I think let's include it. We can always spike on weird, experimental stuff on PRs by overriding the installMoby func to do an alternate installation, ignoring the value in mobyVersion

**Comment by aw-was-here** [comment]
Just a heads up that this pretty much broke the ability for shell scripts to determine what features the command line supports compared to other 'docker' commands. For example, if the client version is => 18, then scripts know that the docker command supports the --cache-from option.  [Note that this is different than the Client API version.]

**Comment by CecileRobertMichon** [comment]
@aw-was-here I'm not sure I understand how this particular PR (making the Moby version configurable) broke the scenario you described above. Can you please explain how this PR is breaking for that scenario?

**Comment by aw-was-here** [comment]
'docker version' is now returning docker client version 3.0.3 in azure pipelines.  Prior it was 18.xxx.

**Comment by CecileRobertMichon** [comment]
@aw-was-here this PR did not change that (it just makes the Moby version configurable). Switching to the Moby container runtime is what changed the version returned: https://github.com/Azure/acs-engine/pull/3896. acs-engine (and now aks-engine) has been delivering moby as the default container runtime since v0.25.0, outlined in the release notes here: https://github.com/Azure/acs-engine/releases/tag/v0.25.0

Docker CE was never supported on acs-engine/aks-engine Linux nodes because of licensing issues (see https://github.com/Azure/acs-engine/issues/4029 for context). The Moby version `3.0.x` that you see when running `docker version` is a Moby build. The problem with moby is that they do not have actual releases and versions - there is no 1:1 to docker CE so any direct comparison isn’t great.

**Comment by aw-was-here** [comment]
Ah, ok, thanks.  I guess we'll likely just yank Azure support for Apache Yetus in Docker mode then since it's unlikely we'll get around to support moby any time soon.

**Pull request merged**